qt/scientific_interfaces/Indirect/ISISEnergyTransfer.cpp

```diff
@@ -92,6 +98,54 @@ void ungroupWorkspace(std::string const &workspaceName) {
   ungroup->execute();
 }
 
+IAlgorithm_sptr loadAlgorithm(std::string const &filename,
+                              std::string const &outputName) {
+  auto loader = AlgorithmManager::Instance().create("Load");
+  loader->initialize();
+  loader->setProperty("Filename", filename);
+  loader->setProperty("OutputWorkspace", outputName);
+  return loader;
+}
+
+void deleteWorkspace(std::string const &name) {
+  auto deleter = AlgorithmManager::Instance().create("DeleteWorkspace");
+  deleter->initialize();
+  deleter->setProperty("Workspace", name);
+  deleter->execute();
+}
+
+double getSampleLog(std::string const &workspaceName,
+                    std::string const &logName, double const &defaultValue) {
+  try {
+    return getADSMatrixWorkspace(workspaceName)->getLogAsSingleValue(logName);
+  } catch (std::exception const &) {
+    return defaultValue;
+  }
+}
+
+double getSampleLog(std::string const &workspaceName,
+                    std::vector<std::string> const &logNames,
+                    double const &defaultValue) {
+  double value(defaultValue);
+  for (auto const &logName : logNames) {
+    value = getSampleLog(workspaceName, logName, defaultValue);
```

Comment:
You're creating an instance of the ADS and retrieving the same workspace for every log name in the vector. It would be more efficient to run `getADSMatrixWorkspace` in this method, and then loop through the vector of logs

Reply:
Good spot 👍